pkg/controller/upload_controller.go

```diff
@@ -98,10 +101,20 @@ func (c *uploadController) enqueueUploadItem(obj interface{}) {
 	log := loggerForUpload(c.logger, req)
 
 	switch req.Status.Phase {
-	case "", pluginv1api.UploadPhaseNew, pluginv1api.UploadPhaseInProgress:
-		// Process New and InProgress Uploads
+	case "", pluginv1api.UploadPhaseNew, pluginv1api.UploadPhaseInProgress, pluginv1api.UploadPhaseUploadError:
+		// Process New and InProgress and UploadError Uploads
 	default:
-		log.Debug("Upload CR is not New or InProgress, skipping")
+		log.Debug("Upload CR is not New or InProgress or UploadError, skipping")
+		return
+	}
+
+	log.Infof("Filtering out the retry upload request which comes in before next retry time")
+	now := c.clock.Now()
+	if now.Unix() < req.Status.NextRetryTimestamp.Unix() {
```

Comment:
since `NextRetryTimestamp` may not be set in the New status, check if it is null as well.